lib/Dialect/HW/HWOps.cpp

```diff
@@ -830,11 +830,14 @@ static LogicalResult verifyOutputOp(OutputOp *op) {
 
 static ParseResult parseSliceTypes(OpAsmParser &p, Type &srcType,
                                    Type &idxType) {
-  ArrayType arrType;
+  Type arrType;
   if (p.parseType(arrType))
```

Comment:
It is unfortunate that `parseType` doesn't for aliased types. Further motivation for getting typedefs supported upstream.